src/core/interpreter.cc

```diff
@@ -1104,4 +1108,16 @@ void InterpreterManager::Reset() {
   VLOG(1) << "InterpreterManager::Reset ended";
 }
 
+void InterpreterManager::Alter(std::function<void(Interpreter*)> modf) {
+  lock_guard guard{reset_mu_};
```

Comment:
`Alter` is atomic for this thread, so why do you need protection here?

Reply:
Just because of the last line 🤷🏻‍♂️  I can actually just call Return for them